src/subclass/interface.rs

```diff
@@ -0,0 +1,241 @@
+    ///
+    /// This will register the type with the type system on the first call and is usually generated
+    /// by the [`glib_object_interface!`] macro.
+    ///
+    /// [`glib_object_interface!`]: ../../macro.glib_object_interface.html
+    fn get_type() -> Type;
+
+    /// Additional type initialization.
+    ///
+    /// This is called right after the type was registered and allows
+    /// interfaces to do additional type-specific initialization, e.g.
+    /// for adding prerequisites.
+    ///
+    /// Optional
+    fn type_init(_type_: &mut InitializingType<Self>) {}
+
+    /// Interface initialization.
+    ///
+    /// This is called after `type_init` and before the first implementor
+    /// of the interface is created. Interfaces can use this to do interface-
+    /// specific initialization, e.g. for installing properties or signals
+    /// on the interface, and for setting default implementations of interface
+    /// functions.
+    ///
+    /// Optional
+    fn interface_init(&mut self) {}
+}
+
+pub trait ObjectInterfaceExt: ObjectInterface {
+    /// Install properties on the interface.
+    ///
+    /// All implementors of the interface must provide these properties.
+    fn install_properties(&mut self, properties: &[Property]) {
+        if properties.is_empty() {
+            return;
+        }
+
+        for property in properties {
+            let pspec = (property.1)(property.0);
+            unsafe {
+                gobject_ffi::g_object_interface_install_property(
+                    self as *mut Self as *mut _,
+                    pspec.to_glib_none().0,
+                );
+            }
+        }
+    }
+
+    /// Add a new signal to the interface.
+    ///
+    /// This can be emitted later by `glib::Object::emit` and external code
+    /// can connect to the signal to get notified about emissions.
+    fn add_signal(&mut self, name: &str, arg_types: &[Type], ret_type: Type) {
+        unsafe {
+            super::types::add_signal(
+                *(self as *mut _ as *mut ffi::GType),
+                name,
+                arg_types,
+                ret_type,
+            );
+        }
+    }
+
+    /// Add a new signal with accumulator to the interface.
+    ///
+    /// This can be emitted later by `glib::Object::emit` and external code
+    /// can connect to the signal to get notified about emissions.
+    ///
+    /// The accumulator function is used for accumulating the return values of
+    /// multiple signal handlers. The new value is passed as second argument and
+    /// should be combined with the old value in the first argument. If no further
+    /// signal handlers should be called, `false` should be returned.
+    fn add_signal_with_accumulator<F>(
+        &mut self,
+        name: &str,
+        arg_types: &[Type],
+        ret_type: Type,
+        accumulator: F,
+    ) where
+        F: Fn(&mut Value, &Value) -> bool + Send + Sync + 'static,
```

Comment:
Maybe use enum { None, TrueHandled, FirstWins, .., Custom(F:Fn())} for accumulator?
Not sure how to handle `g_application_handle_local_options_accumulator` etc,
maybe define trait to get function pointer?

Reply:
An enum would have the same problem that we had with `Option` for optional callbacks, and `Fn` is a trait for giving a function pointer and passing some data around ;)

What exactly do you see wrong with the way how the function is defined? Also this is how we had it before for subclasses already, no change here.

If you'd like to have a different API for that, maybe let's handle that in a separate PR?

Reply:
`g_application_handle_local_options_accumulator` would be

```rust
|return_acc, handler_return| {
  let value = handler_return.get::<u32>().unwrap();
  *return_acc = value.to_value();
  value < 0
}
```

Reply:
Yes can handled later, just note about extension